### PR TITLE
Refactor hosting model application

### PR DESCRIPTION
**Pull request by sirh3e**

### CODE REVIEW

File: src/Microsoft.Tye.Hosting/Model/EnvironmentVariable.cs [hunk 1]
```diff
@@ -27,5 +27,11 @@ public void Deconstruct(out string name, out string? value)
             name = Name;
             value = Value;
         }
+
+        public void Deconstruct(out string name, out string? value, out EnvironmentVariableSource? source)
```
Comment (JunTaoLuo) [inline src/Microsoft.Tye.Hosting/Model/EnvironmentVariable.cs:31]: Where is this used?

File: src/Microsoft.Tye.Hosting/Model/Application.cs [hunk 2]
```diff
@@ -33,19 +33,16 @@ public void PopulateEnvironment(Service service, Action<string, string> set, str
         {
             var bindings = ComputeBindings(service, defaultHost);
 
-            if (service.Description.Configuration != null)
```
Comment (JunTaoLuo) [inline src/Microsoft.Tye.Hosting/Model/Application.cs:36]: Why is this check removed?

**Review by JunTaoLuo — APPROVED** [review]
Two minor questions.

**Comment by jkotalik** [comment]
Going to close for now, feel free to reopen and fix the remaining comments.

**Pull request closed without merging**